.gitlab-ci.yml

```diff
@@ -241,11 +241,21 @@ pkg:nix:
   stage: test
   variables:
     GIT_STRATEGY: none
+    # By default we use coq.cachix.org as an extra substituter but this can be overridden
+    EXTRA_SUBSTITUTERS: "${EXTRA_SUBSTITUTERS-https://coq.cachix.org}"
+    EXTRA_PUBLIC_KEYS: "${EXTRA_PUBLIC_KEYS-coq.cachix.org-1:Jgt0DwGAUo+wpxCM52k2V+E0hLoOzFPzvg94F65agtI=}"
+
   dependencies: [] # We don't need to download build artifacts
   before_script: [] # We don't want to use the shared 'before_script'
   script:
+    # Use current worktree as tmpdir to allow exporting artifacts in case of failure
     - export TMPDIR=$PWD
-    - nix-build "$CI_PROJECT_URL/-/archive/$CI_COMMIT_SHA.tar.gz" -K
+    # Install Cachix as documented at https://github.com/cachix/cachix
+    - nix-env -if https://github.com/cachix/cachix/tarball/master --substituters https://cachix.cachix.org --trusted-public-keys cachix.cachix.org-1:eWNHQldwUO7G2VkjpnjDbWwy4KQ/HNxht7H4SSoMckM=
```

Comment:
This key and the other one should also be in variables.

Reply:
I don't see why you would want to replace the official cache for this program without replacing the source for the program as well. This might be over-thinking it.

Reply:
It has nothing to do about replacing, it's about moving the magic value out of the script.